Ingress restrictions

devzspy opened the pull request:
Addition of the ingress-restrictions to allow edge hosts to reach middle hosts on port 2222. 

Resolves #70

devzspy committed:
feat(ansible): Add role for managing ingress restrictions

Introduces a new Ansible role `ingress-restrictions` with tasks to manage
ingress restrictions between edge and middle hosts. Also includes necessary
files such as handlers, vars, defaults, and README.

r00tkillah commented on external/sketch/ansible/roles/ingress-restrictions/README.md:
```diff
@@ -0,0 +1,38 @@
+Role Name
+=========
+
+A brief description of the role goes here.
```
Please provide a brief description here

r00tkillah commented on external/sketch/ansible/roles/ingress-restrictions/README.md:
this is the template. fill it out

devzspy replied:
It was filled out in the latest commit

r00tkillah replied:
squash the two commits?

r00tkillah commented on external/sketch/ansible/roles/ingress-restrictions/meta/main.yml:
fill out

r00tkillah reviewed (dismissed):
please fill out the boilerplate stuff. not ansible smart enough to render opinion on the ansible part

devzspy commented on external/sketch/ansible/roles/ingress-restrictions/defaults/main.yml:
Updated

tophertimzen commented:
Does this commit really resolve #70? The proxies are still exposing 2222 to the internet, aren't they? and scanners can correlate the SSH fingerprint.

devzspy commented on external/sketch/ansible/roles/ingress-restrictions/vars/main.yml:
Updated

devzspy commented on external/sketch/ansible/roles/ingress-restrictions/meta/main.yml:
```diff
@@ -0,0 +1,52 @@
+galaxy_info:
```
Updated

tophertimzen commented on external/sketch/ansible/roles/ingress-restrictions/README.md:
```diff
@@ -0,0 +1,24 @@
+Requirements
+------------
+
+Once you've ran this in your Sketch infrastructure, you will need to grab the IP Addresses or IP Address range from your middles and add them to your OCI/AWS infrastructure manually.
```
How do you do this? This seems to be a critical part of fixing #70

devzspy replied:
It may not fully resolve #70 given we had to setup 1:1:1 correlation of proxy, middle, edge in an engagement. However, outside of that, we are blocking 2222 on all of Sketch using this. The only manual part is taking our middles and throwing them into OCI/AWS to allow there from middles and block elsewhere.

tophertimzen replied:
If it doesn't fully resolve #70, is there a new ticket for the remainder of the work? Without it fully resolved, there is still an opsec leak unless that manual process is performed - which should be clearly documented for now.

tophertimzen replied:
https://github.com/redteaminfra/redteam-infra/issues/70#issuecomment-2101016861

devzspy replied:
I've broken up the issue into two parts and renamed this one. #70 is now only for the work handled by this role while #114 covers the remainder work and enchancements.

willk commented on external/sketch/ansible/roles/ingress-restrictions/tasks/main.yml:
I think the port should be a variable, 2222 isn't set in stone.

willk reviewed (changes requested):
Please change it so the port can be a variable. I would also recommend allow for multiple ports to facilitate #114

devzspy committed:
Merge branch 'redteaminfra:master' into ingress-restrictions

willk reviewed (approved):
Looks good with the boilerplate

devzspy committed:
feat(ingress): make backflip port configurable via variable

Replace hardcoded 2222 with {{ backflip_port }} default. Addresses review feedback to avoid hardcoding the port number for future flexibility (e.g., multiple backflip ports per #114).

willk reviewed (approved):
Looks like a valid ansible role
